test/client/fixtures/utils/url-test.js

```diff
@@ -42,6 +41,38 @@ test('resolveUrlAsDest', function () {
     strictEqual(urlUtils.resolveUrlAsDest('http://g.tbcdn.cn/??kissy/k/1.4.2/seed-min.js'), 'http://g.tbcdn.cn/??kissy/k/1.4.2/seed-min.js');
 });
 
+test('isSupportedProtocol', function () {
+    ok(!urlUtils.isSupportedProtocol('#42'));
+    ok(!urlUtils.isSupportedProtocol(' data:asdasdasdasdasdasd'));
+    ok(!urlUtils.isSupportedProtocol('chrome-extension://google.com/image.png'));
```

Comment:
I've asked to add tests for all scenarios and positive scenarios as well:
`http://example.org`,
`https://example.org`,
`//example.org`,
`/some/path`
`somepath`
etc.